security/filter/security-filter-login/src/main/java/org/codice/ddf/security/filter/login/LoginFilter.java

```diff
@@ -105,12 +105,24 @@
 
     private Crypto signatureCrypto;
 
-    private DocumentBuilder docBuilder;
-
     private Validator assertionValidator = new SamlAssertionValidator();
 
     private final Object lock = new Object();
 
+    private static final ThreadLocal<DocumentBuilder> BUILDER =
+            new ThreadLocal<DocumentBuilder>() {
+                @Override
+                protected DocumentBuilder initialValue() {
+                    try {
+                        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                        factory.setNamespaceAware(true);
+                        return factory.newDocumentBuilder();
+                    } catch (ParserConfigurationException ex) {
+                        throw new IllegalArgumentException("Unable to create new DocumentBuilder", ex);
+                    }
```

Comment:
This exception should never happen as things are configured here.

http://docs.oracle.com/javase/7/docs/api/javax/xml/parsers/ParserConfigurationException.html